prow/plugins/buildifier/buildifier.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// buildifier defines a Prow plugin that runs buildifier over modified BUILD,
+// WORKSPACE, and skylark (.bzl) files in pull requests.
+package buildifier
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/bazelbuild/buildtools/build"
```

Comment:
Needs to be vendored.

Reply:
See my comment above, what's the process?

Reply:
I am stumbling upon various issues when trying to use `godep` :(

https://github.com/kubernetes/test-infra/issues/4868
https://github.com/kubernetes/test-infra/issues/4867

I played around with `dep` and adding a new dependency seems to work great with `dep ensure` but the transition from Godep to dep is wonky.

Reply:
If you want to switch this repo to use `go dep` instead of `godep`, everyone would be happier I think.

Reply:
Note that `dep` is not yet part of the go toolchain and likely not ready for prime time but I started using it during the past month and it has worked great for me. I am all for changing to use it but, as I said in my previous comment, the transition from Godep to dep is not great. I didn't introspect closely my new deps yesterday but it was a massive diff likely because the current state of our vendor dir is not ideal. I wasn't also sure if I got the versions correctly and wasn't creating subtle breakages in different tools (kubetest, janitor, prow, etc). Maybe this is an argument in favor of splitting test-infra to more scoped repositories.